filebeat/input/filestream/fswatch.go

```diff
@@ -142,10 +142,21 @@ func (w *fileWatcher) watch(ctx unison.Canceler) {
 		}
 
 		if prevInfo.ModTime() != info.ModTime() {
-			select {
-			case <-ctx.Done():
-				return
-			case w.events <- writeEvent(path, info):
+			fmt.Println(prevInfo.Size(), info.Size())
+			// if a file has been changed according to modtime but the size is the same
+			// it is still considered truncation.
```

Comment:
This means a simple `$ touch <file>` will recollect said file? Does the log input behave the same?

I kinda like the idea of being able to resend a file by touching it locally. But I wonder if we want to make this behavior optional? Anyways, we should document it, as it also addresses a not so uncommon use-case (reindex existing file).

Reply:
Yes, you are correct. It is a new feature. I would like to document it and enable it by default unless people complain about it.
I was thinking about use cases when people would just run `touch` on their log files or what application would just change the modification time of a log without adding or removing contents. So far I have nothing, but I am afraid there might be some exotic log writing application on Windows I am not aware of.